### PR TITLE
[stable29] ci(build): find a fitting NC version fallback

### PR DESCRIPTION
**Pull request by blizzz**
Addresses https://github.com/nextcloud/.github/issues/340 so should go upstream if it works :crossed_fingers: 

Parsing by grep is maybe not the most elegant way though, but perhaps most pragmatic way here.

### CODE REVIEW

File: .github/workflows/appstore-build-publish.yml [hunk 1]
```diff
@@ -130,9 +130,18 @@ jobs:
...
+        id: server-checkout-by-phpv
+        run: |
+          DOWNLOADURL=$(curl https://raw.githubusercontent.com/nextcloud-releases/updater_server/master/config/config.php | grep -B3 "minPHPVersion' => '{{ steps.php-versions.outputs.php-min }}'"  | head -n 3 | grep downloadUrl | grep -oE "https://[^']*")
+          wget --quiet -O nextcloud.zip "${DOWNLOADURL}"
+          unzip nextcloud.zip
+
       - name: Checkout server master fallback
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
-        if: ${{ steps.server-checkout.outcome != 'success' }}
+        if: ${{ steps.server-checkout-by-phpv.outcome != 'success' }}
```
Comment (juliushaertl) [inline .github/workflows/appstore-build-publish.yml:144]: Shouldn't we also skip this if server-checkout was successful? My assumption would be that if server-checkout-by-phpv is skipped we still run the checkout from git then

Comment (blizzz) [reply .github/workflows/appstore-build-publish.yml:144]: i thought it would not qualify in this case

Comment (blizzz) [reply .github/workflows/appstore-build-publish.yml:144]: We could also include *skipped* as value in this check.

**Comment by blizzz** [comment]
Now reality (aka 29 release) caught up with me :sweat_smile: next time :wink:

**Pull request closed without merging**